main-settings/src/main/scala/sbt/Reference.scala

```diff
@@ -47,7 +51,7 @@ final case class RootProject(build: URI) extends ProjectReference
 final case object LocalRootProject extends ProjectReference
 
 /** Identifies the project for the current context. */
-final case object ThisProject extends ProjectReference
+final case object ThisProject extends ScopeFilterProjectReference
```

Comment:
Is this binary compatible?

Reply:
No. But what do you think of the idea?

Reply:
Is this trying to save `ThisProject`, or push it into scope filtering only? I guess I'm not seeing the point.

Reply:
It's trying to fix Scope's `resolveProjectBuild` and `resolveProjectRef` to no
longer mishandle `ThisProject`.

But to do so it's removing `ThisProject` as a legitimate member of the `ProjectReference` sealed family. Thinking about it over-night I think it is a legitimate member, should remain so, and the real fix for 2.0.0 is to change resolveProjectBuild and resolveProjectRef.